File: VB/GridWithFilterPanelButtons/CustomGridControl/CustomGridControlView.vb
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Views.Grid
Imports System
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Linq

Namespace GridWithFilterPanelButtons
    Public Class CustomGridControlView
        Inherits GridView


        Private customFilterPanelButtons_Renamed As ObservableCollection(Of EditorButton)

        Public Sub New()
            AddHandler CustomFilterPanelButtons.CollectionChanged, AddressOf OnCollectionChanged
        End Sub

        Public Sub New(ByVal grid As GridControl)
            MyBase.New(grid)
        End Sub

        Private Sub OnCollectionChanged(ByVal sender As Object, ByVal e As System.Collections.Specialized.NotifyCollectionChangedEventArgs)
            OnPropertiesChanged()
        End Sub

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            RemoveHandler CustomFilterPanelButtons.CollectionChanged, AddressOf OnCollectionChanged
            MyBase.Dispose(disposing)
        End Sub

        Protected Overrides ReadOnly Property ViewName() As String
            Get
                Return "CustomGridControlView"
            End Get
        End Property

        <DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
        Public ReadOnly Property CustomFilterPanelButtons() As ObservableCollection(Of EditorButton)
            Get
                If customFilterPanelButtons_Renamed Is Nothing Then
                    customFilterPanelButtons_Renamed = New ObservableCollection(Of EditorButton)()
                End If
                Return customFilterPanelButtons_Renamed
            End Get
        End Property
    End Class
End Namespace

File: CS/GridWithFilterPanelButtons/CustomGridControl/CustomGridControlView.cs
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Views.Grid;
using System;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;

namespace GridWithFilterPanelButtons
{
    public class CustomGridControlView : GridView
    {
        ObservableCollection<EditorButton> customFilterPanelButtons;

        public CustomGridControlView()
        {
            CustomFilterPanelButtons.CollectionChanged += OnCollectionChanged;
        }

        public CustomGridControlView(GridControl grid) : base(grid) { }

        private void OnCollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
        {
            OnPropertiesChanged();
        }

        protected override void Dispose(bool disposing)
        {
            CustomFilterPanelButtons.CollectionChanged -= OnCollectionChanged;
            base.Dispose(disposing);
        }

        protected override string ViewName => "CustomGridControlView";

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public ObservableCollection<EditorButton> CustomFilterPanelButtons {
            get {
                if(customFilterPanelButtons == null)
                    customFilterPanelButtons = new ObservableCollection<EditorButton>();
                return customFilterPanelButtons;
            }
        }
    }
}
